cmd/data-node/commands/start/node.go

```diff
@@ -70,11 +70,15 @@ type NodeCommand struct {
 	VersionHash string
 }
 
-func (l *NodeCommand) Run(cfgwatchr *config.Watcher, vegaPaths paths.Paths, args []string) error {
+func (l *NodeCommand) Run(ctx context.Context, cfgwatchr *config.Watcher, vegaPaths paths.Paths, args []string) error {
 	l.configWatcher = cfgwatchr
 
 	l.conf = cfgwatchr.Get()
 	l.vegaPaths = vegaPaths
+	if l.cancel != nil {
+		l.cancel()
+	}
+	l.ctx, l.cancel = context.WithCancel(ctx)
```

Comment:
IIRC it's generally considered bad form to put context in structs right? they're supposed to be associated with the call stack rather than stored.. I figure it's probably OK in the command structs further above since they're basically structs to put function parameters in but I can't quite figure out the need here..

Reply:
e.g. https://go.dev/blog/context-and-structs

Reply:
Most commands have a context field. It's currently the only way to pass the context down from the "higher" command. It's not the prettiest solution, but it's used all over (e.g. vega/commands/node/node.go has `ctx`  and `cancel` fields for this reason). It's the best way to ensure that, hen the `Run` function returns (and the context is cancelled), this is propagated throughout, and -in this case- we know it's time to clean up.